compiler/rustc_hir_typeck/src/pat.rs

```diff
@@ -2409,7 +2409,7 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         min_len: u64,
     ) -> (Option<Ty<'tcx>>, Ty<'tcx>) {
         let len = match len.eval(self.tcx, self.param_env, span) {
-            Ok(val) => val
+            Ok((_, val)) => val
```

Comment:
debug assert that the type is `usize`?

Reply:
I think this will probably ICE under `adt_const_params` since we can have aliases for the `type_of` of anon consts which we sometimes use as the `Ty` here. (which I think is a bit gross but lol this PR was huge already)

Reply:
Added a `FIXME` I'll try get to this soon